### ISSUE TITLE
Cannot assign dict in list comprehension to TypedDict

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#1008]
```python
from typing import List

from typing_extensions import TypedDict


MessageTypeDef = TypedDict("MessageTypeDef", {"Id": str, "Handle": str})

msgs = [{"Id": "1", "Handle": "2"}]
typed: MessageTypeDef = {"Id": msgs[0]["Id"], "Handle": msgs[0]["Handle"]}
list1: List[MessageTypeDef] = [{"Id": msgs[0]["Id"], "Handle": msgs[0]["Handle"]}]
list2: List[MessageTypeDef] = [{"Id": msg["Id"], "Handle": msg["Handle"]} for msg in msgs]
```

`list1` is fine, but `list2` errors when I wasn't expecting it:
```
  11:31 - error: Expression of type "list[dict[str, str]]" cannot be assigned to declared type "List"
  TypeVar "_T" is invariant
    "dict[str, str]" is incompatible with "MessageTypeDef"
```

pyright 1.1.65

**Comment by erictraut** [comment]
Thanks for reporting the issue.

I've extended support for bidirectional type inference to list comprehensions, allowing this case to work as expected. This will be in the next version of Pyright and Pylance.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.67, which I just published. It will also be included in next week's release of Pylance.
